cachingClient/src/index.js

```diff
@@ -116,6 +116,11 @@ class CachingClient extends EventDispatcher{
             }
         } else if (!cachedResponse && (method == CachingClient.PUT || method == CachingClient.DELETE)) {
             throw new Error(`Can't do a ${request.method} without previously having information in the Cache about the resource`);
+        } else if (cachedResponse && (method == CachingClient.PUT || method == CachingClient.DELETE)) {
+            const ifMatch = cachedResponse.headers.get('ETag');
+            const ifUnmodifiedSince = cachedResponse.headers.get('Last-Modified');
+            request.headers.append('If-Match', ifMatch);
```

Comment:
These two lines need to be conditional upon the headers being included in the response. That is, something like `if 'ETag' in cachedResponse.headers`. Another issue - some libraries will automatically lowercase the header names - do you know if axios does? You'll need to confirm that these strings work and their lowercase versions aren't required by the library.

Reply:
Checking a little bit seems like Axios does lower case the headers but seems like only for the responses  (axios/axios#413 and https://github.com/axios/axios/blob/c4300a88cf994b9c4b88b065806edb98705a2c5d/README.md#response-schema).

For context, seems like the spec says the headers are case-insensitive but some implementations could capitalize the headers: https://stackoverflow.com/a/5259004

So should I ensure that headers are lower cased everywhere?

Will check the validation to append actual values to the request :+1:

Reply:
HTTP 1.1 and 2.0 disagree here - 2.0 standardizes on lower-case. I think... let's try lower case and hope that axios/browser deal with it.